spinalcordtoolbox/deepseg_gm/deepseg_gm.py

```diff
@@ -113,6 +113,50 @@ def pad(self, image):
         return padded
 
 
+class StandardizationTransform(object):
+    """This transformation will standardize the volume
+    according to the specified mean/std.dev.
+    """
+    def __init__(self, mean, std):
+        """Constructor for the normalization transformation.
+
+        :param mean: the mean parameter
+        :param std: the standar deviation parameter
+        """
+        self.mean = mean
+        self.std = std
+
+    def __call__(self, volume):
+        """This method will enable the function call for the
+        class object.
+
+        :param volume: the volume to be normalized.
+        """
+        volume -= self.mean
+        volume /= self.std
+        return volume
+
+
+class VolumeStandardizationTransform(object):
+    """This transformation will standardize the volume with
+    the parameters estimated from the volume itself.
+    """
+
+    def __call__(self, volume):
+        """This method will enable the function call for the
+        class object.
+
+        :param volume: the volume to be normalized.
+        """
+        volume_mean = volume.mean()
+        volume_std = volume.std()
+
+        volume -= volume_mean
+        volume /= volume_std
+
+        return volume
+
+
```

Comment:
A functor seems overkill.

Reply:
This is the usual pattern for transformations [in PyTorch for instance](https://github.com/pytorch/vision/blob/master/torchvision/transforms/transforms.py#L120), even for the same purpose, so I don't think it's overkill, but if you want me to change, please tell me what you prefer so I can change it.